component/src/main/java/org/ballerinalang/compiler/FTPMonitorServiceCompilerPlugin.java

```
@@ -0,0 +1,135 @@
+        if (NodeKind.RECORD_LITERAL_EXPR.equals(configurationExpression.getKind())) {
+            BLangRecordLiteral recordLiteral = (BLangRecordLiteral) configurationExpression;
+            boolean isNonEmptyPath = false;
+            boolean isNonEmptyHost = false;
+            for (BLangRecordLiteral.BLangRecordKeyValue config : recordLiteral.getKeyValuePairs()) {
+                final String key = ((BLangSimpleVarRef) config.getKey()).variableName.value;
+                if (config.getValue() instanceof BLangLiteral) {
+                    final Object value = ((BLangLiteral) config.getValue()).getValue();
+                    switch (key) {
+                        case FtpConstants.ENDPOINT_CONFIG_PATH:
+                            if (value != null) {
+                                if (!value.toString().isEmpty()) {
+                                    isNonEmptyPath = true;
+                                }
+                            }
+                            break;
+                        case FtpConstants.ENDPOINT_CONFIG_HOST:
+                            if (value != null) {
+                                if (!value.toString().isEmpty()) {
+                                    isNonEmptyHost = true;
+                                }
+                            }
+                            break;
+                        default:
+                            // Do nothing.
+                    }
+                }
+            }
+            if (!isNonEmptyPath) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, endpointNode.getPosition(),
+                        "Cannot create FTP Listener without path.");
+            }
+            if (!isNonEmptyHost) {
+                dlog.logDiagnostic(Diagnostic.Kind.ERROR, endpointNode.getPosition(),
+                        "Cannot create FTP Listener without host.");
+                throw new BallerinaException("Cannot create FTP Listener without host.");
```

if ((BLangLiteral) config.getValue()).getValue() returns null we should throw this exception right ?

---

I have updated the code to throw error before the switch statement if the value is null.